library/alloc/src/collections/btree/node.rs

```diff
@@ -1741,5 +1734,14 @@ unsafe fn slice_remove<T>(slice: &mut [MaybeUninit<T>], idx: usize) -> T {
     }
 }
 
+/// Works like `dst.copy_from_slice(src)` but does not require `T` to be `Copy` nor `Clone`.
+/// Useful to implement a move, where you consider `src` uninitialized afterwards.
+fn copy_from_slice<T>(dst: &mut [MaybeUninit<T>], src: &[MaybeUninit<T>]) {
```

Comment:
I think this should be called "move_from_slice" and the argument order made consistent with copy_nonoverlapping.

Reply:
Do you really want the order of a `move_from_slice` to be the inverse of [slice::copy_from_slice](https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice)? And should edges use `slice::copy_from_slice` (as in the current PR) or this `move_from_slice`?

Reply:
copy_from_slice is a method, and this is a free function, so conventions/naming are a bit different imo.

move_to_slice would be great, excellent point.